flash/data/data_pipeline.py

```diff
@@ -0,0 +1,612 @@
+
+                # if no additional funmc available: remove wrapper
+                if model.transfer_batch_to_device.is_empty():
+                    model.transfer_batch_to_device = model.transfer_batch_to_device.func
+
+            if device_collate is None:
+                device_collate = self._do_nothing_collate
+
+            loader_name = f'{_STAGES_PREFIX[stage]}_dataloader'
+
+            dataloader, whole_attr_name = self._get_dataloader(model, loader_name)
+
+            if dataloader is None:
+                continue
+
+            if isinstance(dataloader, _PatchDataLoader):
+                dataloader = dataloader()
+            elif isinstance(dataloader, Callable):
+                dataloader = dataloader()
+
+            if isinstance(dataloader, Sequence):
+                was_seq = True
+            else:
+                dataloader = [dataloader]
+                was_seq = False
+
+            for idx, loader in enumerate(dataloader):
+                if isinstance(loader, DataLoader):
+                    dl_args = {k: v for k, v in vars(loader).items() if not k.startswith("_")}
+
+                    if isinstance(dl_args['collate_fn'], _PreProcessor):
+                        dl_args['collate_fn'] = dl_args['collate_fn']._original_collate_fn
+                        del dl_args["batch_sampler"]
+                        loader = type(loader)(**dl_args)
+
+                dataloader[idx] = loader
+
+            if not was_seq:
+                dataloader = dataloader[0]
+
+            if isinstance(dataloader, DataLoader):
+                dataloader = _PatchDataLoader(dataloader)
+
+            self._set_loader(model, whole_attr_name, dataloader)
+
+    @staticmethod
+    def _detach_postprocess_from_model(model: 'Task'):
+
+        if hasattr(model.predict_step, '_original'):
+            # don't delete the predict_step here since we don't know
+            # if any other pipeline is attached which may rely on this!
+            model.predict_step = model.predict_step._original
+
+    def _generate_callable_auto_dataset(
+        self, data: Union[Iterable, Any], running_stage: RunningStage = None
+    ) -> Callable:
+
+        def fn():
+            return self._generate_auto_dataset(data, running_stage=running_stage)
+
+        return fn
+
+    def _generate_auto_dataset(self, data: Union[Iterable, Any], running_stage: RunningStage = None) -> AutoDataset:
+        return AutoDataset(data=data, data_pipeline=self, running_stage=running_stage)
+
+    def to_dataloader(
+        self, data: Union[Iterable, Any], auto_collate: Optional[bool] = None, **loader_kwargs
+    ) -> DataLoader:
+        if 'collate_fn' in loader_kwargs:
+            if auto_collate is not None:
+                raise MisconfigurationException('auto_collate and collate_fn are mutually exclusive')
+
+        else:
+            if auto_collate is None:
+                auto_collate = True
```

Comment:
this looks strange, you just revert it?

Reply:
Another PR. There might be a better way to handle default collate.